idb/postgres/postgres.go

```diff
@@ -2589,8 +2589,9 @@ func (db *IndexerDb) Assets(ctx context.Context, filter idb.AssetsQuery) (<-chan
 		partNumber++
 	}
 	if filter.Name != "" {
+		queryName := encoding.SanitizeNullForQuery(filter.Name)
```

Comment:
To solve the problem I mentioned, we just have to remove this. Now the escaped format will be accepted and if you submit a search with a null character you'll get a malformed query error.

Reply:
Huh? I just got convinced we need to do this. I'd expect we need to do at least the regular forward EscapeNulls(), but then by integration test we can see that lib/pq isn't escaping that right so we need to double escape it; so what chain of events lead to us not needing to do any escaping at all?